### PR TITLE
Session

### PR DESCRIPTION
**Pull request by joredudiaz**
## Summary

- save session

## Jira Story

https://trello.com/c/RMsLYMuB

## Screenshot

session

### CODE REVIEW

File: wolox-books/src/app/components/navbar/navbar.component.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<div class="nav">
+  <img src="../../assets/logo-wolox.png" alt="wolox-logo" class="logo">
+  <a class="nav-title" (click)="closeSession()">Logout</a>
```
Comment (FrankIglesias) [inline wolox-books/src/app/components/navbar/navbar.component.html]: This should be a button

File: wolox-books/src/app/components/navbar/navbar.component.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
+@import '../../../colors.scss';
+
+.nav {
+  align-items: center;
+  background-color: $white;
+  border-top: solid $cerulean 6px;
+  box-shadow: 0 0 10px 0 rgba(0,0,0,0.5);
```
Comment (FrankIglesias) [inline wolox-books/src/app/components/navbar/navbar.component.scss]: use variables for this rgba

File: wolox-books/src/app/components/navbar/navbar.component.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  display: flex;
+  grid-area: header;
+  height: 64px;
+  justify-content: space-between;
+  padding: 0 246px;
+  position: sticky;
```
Comment (FrankIglesias) [inline wolox-books/src/app/components/navbar/navbar.component.scss]: fixed

File: wolox-books/src/app/components/navbar/navbar.component.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  z-index: 1;
+}
+
+.nav-logo {
+  width: 200px;
+}
+
+.nav-title {
+  font-size: 18px;
+  font-family: 'roboto';
```
Comment (FrankIglesias) [inline wolox-books/src/app/components/navbar/navbar.component.scss]: Define an equalizer to apply this font family to all texts

File: wolox-books/src/app/components/navbar/navbar.component.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { Component, OnInit } from '@angular/core';
+import { SessionService } from 'src/app/session.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-navbar',
```
Comment (FrankIglesias) [inline wolox-books/src/app/components/navbar/navbar.component.ts]: remove prefix from components using --no-prefix

File: wolox-books/src/app/components/navbar/navbar.component.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  templateUrl: './navbar.component.html',
+  styleUrls: ['./navbar.component.scss']
+})
+export class NavbarComponent implements OnInit {
+
+  constructor(private sessionService: SessionService, private router: Router) { }
+
+  ngOnInit(): void {
```
Comment (FrankIglesias) [inline wolox-books/src/app/components/navbar/navbar.component.ts]: remove unnecessary functions

File: wolox-books/src/app/components/navbar/navbar.component.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  closeSession() {
+    this.sessionService.deleteSession();
+    this.router.navigate(['login']);
+  }
+
```
Comment (FrankIglesias) [inline wolox-books/src/app/components/navbar/navbar.component.ts]: remove extra line

File: wolox-books/src/app/screens/book/book-routing.module.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { ListComponent } from './components/list/list.component';
+import { BookComponent } from './book.component';
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+
+
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/book-routing.module.ts]: remove extra line

File: wolox-books/src/app/screens/book/book-routing.module.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+const routes: Routes = [
+  {
+    path: 'books', component: BookComponent,
+    children: [
+      { path: 'list', component: ListComponent }
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/book-routing.module.ts]:
```suggestion
      { path: '', component: ListComponent }
```

File: wolox-books/src/app/screens/book/book.component.scss [hunk 10]
```diff
@@ -0,0 +1,7 @@
+@import '../../../colors.scss';
+
+:host {
+  background-color: $wild-sand;
+  height: 100vh;
+  width: 100%;
+}
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/book.component.scss]: add eol here

File: wolox-books/src/app/screens/book/book.component.scss [hunk 11]
```diff
@@ -0,0 +1,7 @@
+@import '../../../colors.scss';
+
+:host {
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/book.component.scss:3]: Define this as a container class attributes

File: wolox-books/src/app/screens/book/book.component.scss [hunk 12]
```diff
@@ -0,0 +1,7 @@
+@import '../../../colors.scss';
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/book.component.scss]: remove file extension here

File: wolox-books/src/app/screens/book/book.component.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-book',
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/book.component.ts]: remove prefix

File: wolox-books/src/app/screens/book/book.component.ts [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  templateUrl: './book.component.html',
+  styleUrls: ['./book.component.scss']
+})
+export class BookComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/book.component.ts]: remove these

File: wolox-books/src/app/screens/book/components/list/list.component.html [hunk 15]
```diff
@@ -0,0 +1,4 @@
+<div class="search">
+  <input type="text" class="input-search" placeholder="Busc&aacute; por t&iacute;tulo del libro...">
+</div>
+<div class="books"></div>
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/components/list/list.component.html]: add EOL

Comment (FrankIglesias) [reply wolox-books/src/app/screens/book/components/list/list.component.html]: Unnecessary div I think

Comment (FrankIglesias) [reply wolox-books/src/app/screens/book/components/list/list.component.html]: Missing association with a model

File: wolox-books/src/app/screens/book/components/list/list.component.scss [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+  background-color: $white;
+  border-radius: 5px;
+  border: 2px solid $mercury;
+  font-family: 'Roboto';
+  font-size: 20px;
+  height: 50px;
+  line-height: 20px;
+  padding: 0 15px;
+  width: 646px;
+}
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/components/list/list.component.scss]: missing eol

File: wolox-books/src/app/screens/book/components/list/list.component.scss [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+  background: url(../../../../../assets/search.png) no-repeat scroll 600px 10px;
+  background-color: $white;
+  border-radius: 5px;
+  border: 2px solid $mercury;
+  font-family: 'Roboto';
+  font-size: 20px;
+  height: 50px;
+  line-height: 20px;
+  padding: 0 15px;
+  width: 646px;
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/components/list/list.component.scss]:
```suggestion
  max-width: 646px;
  width: 100%;
```

File: wolox-books/src/app/screens/book/components/list/list.component.scss [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+.search {
+  padding: 32px 0 45px 0;
+}
+
+.input-search {
+  background: url(../../../../../assets/search.png) no-repeat scroll 600px 10px;
+  background-color: $white;
+  border-radius: 5px;
+  border: 2px solid $mercury;
+  font-family: 'Roboto';
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/components/list/list.component.scss]: define this in an equalizer

File: wolox-books/src/app/screens/book/components/list/list.component.scss [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+
+:host {
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  padding: 0 235px;
+}
+
+.search {
+  padding: 32px 0 45px 0;
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/components/list/list.component.scss]:
```suggestion
  padding: 32px 0 45px;
```

File: wolox-books/src/app/screens/book/components/list/list.component.ts [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+  templateUrl: './list.component.html',
+  styleUrls: ['./list.component.scss']
+})
+export class ListComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/components/list/list.component.ts]: remove these

File: wolox-books/src/app/session.service.ts [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+  constructor() { }
+
+  createSession(token: string): void {
+    localStorage.setItem(this.TOKEN_KEY, token);
+  }
+
+  hasSession(): boolean {
+    let token = null;
+
+    if (localStorage.length > 0) {
```
Comment (FrankIglesias) [inline wolox-books/src/app/session.service.ts]: do you need this if?

File: wolox-books/src/app/session.service.ts [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+      token = localStorage.getItem(this.TOKEN_KEY);
+    }
+
```
Comment (FrankIglesias) [inline wolox-books/src/app/session.service.ts]: remove line

File: wolox-books/src/colors.scss [hunk 23]
```diff
@@ -4,4 +4,5 @@ $white: #fff;
 $cerulean: #00ADEE;
 $silver-chalice: #A8A8A8;
 $red: #f00;
-$olivine: #a0bf67
+$olivine: #a0bf67;
```
Comment (FrankIglesias) [inline wolox-books/src/colors.scss]: use uppercase for colors same above

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: wolox-books/src/app/session.service.ts (hunk 21 above)
Comment (joredudiaz) [reply wolox-books/src/app/session.service.ts]: nup

File: wolox-books/src/app/components/navbar/navbar.component.html [hunk 24]
```diff
@@ -0,0 +1,4 @@
+<div class="nav">
+  <img src="../../assets/logo-wolox.png" alt="wolox-logo" class="logo">
+  <button class="nav-title" (click)="closeSession()">Logout</button>
```
Comment (FrankIglesias) [inline wolox-books/src/app/components/navbar/navbar.component.html]: missing type here

File: wolox-books/src/app/components/navbar/navbar.component.scss [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
```
Comment (FrankIglesias) [inline wolox-books/src/app/components/navbar/navbar.component.scss]: remove this line

File: wolox-books/src/app/screens/book/components/list/list.component.scss [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+  text-align: center;
+  width: 100%;
+}
+
+.input-search {
+  background: url(../../../../../assets/search.png) no-repeat scroll 97% 10px;
+  background-color: $white;
+  border-radius: 5px;
+  border: 2px solid $mercury;
+  font-family: 'Roboto';
```
Comment (FrankIglesias) [inline wolox-books/src/app/screens/book/components/list/list.component.scss]: remove this line

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: wolox-books/src/app/components/navbar/navbar.component.html (hunk 24 above)
Comment (joredudiaz) [reply wolox-books/src/app/components/navbar/navbar.component.html]: I don't understand, what type?

Comment (FrankIglesias) [reply wolox-books/src/app/components/navbar/navbar.component.html]:
```suggestion
  <button type="button" class="nav-title" (click)="closeSession()">Logout</button>
```

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**